dao/src/main/java/org/thingsboard/server/dao/device/DeviceServiceImpl.java

```diff
@@ -388,84 +388,84 @@ public PageData<Device> findDevicesByTenantIdAndTypeAndEmptyOtaPackage(TenantId
                                                                            PageLink pageLink) {
         log.trace("Executing findDevicesByTenantIdAndTypeAndEmptyOtaPackage, tenantId [{}], deviceProfileId [{}], type [{}], pageLink [{}]",
                 tenantId, deviceProfileId, type, pageLink);
-        validateId(tenantId, INCORRECT_TENANT_ID + tenantId);
-        validateId(tenantId, INCORRECT_DEVICE_PROFILE_ID + deviceProfileId);
+        validateId(tenantId, id -> INCORRECT_TENANT_ID + id);
+        validateId(tenantId, id -> INCORRECT_DEVICE_PROFILE_ID + id);
```

Comment:
There should be `deviceProfileId` check instead of `tenantId` at the second check. This is not your fault, but please fix it.